api/filters/sharing_instance_filter.go

```diff
@@ -0,0 +1,168 @@
+/*
```

Comment:
overall comment, consider adding more debug and info logs in this code so that it will be more easy to debug and learn what did happen in the future.